File: test_ex_1.java

public class test_ex_1 {

    public static int main(String mot) {
        int count = 0;
        String voyelles = "aeiouy";

        for (int i = 0; i < mot.length(); i++) {
            char c = mot.charAt(i);
            if (voyelles.indexOf(c) != -1) {
                count++;
            }
        }

        return count;
    }

    public static void main(String[] args) {
        String mot = "hello world";
        System.out.println(voyelles(mot));
    }
}

File: test_ex_1.js

function voyelles(s) {
    let count = 0;
    let voyelles = "aeiouy";

    for (let i = 0; i < s.length; i++) {
        let c = s[i];

        if (voyelles.includes(c)) {
            count++;
        }
    }
    return count;
}

let s = "hello world";
console.log(voyelles(s));